### PR TITLE
Bundle Pure ESM Packages into `/vendors/*.js` as CommonJS

### PR DESCRIPTION
**Pull request by sosukesuzuki**
## Description

Bundles external Pure ESM Packages to CJS (https://github.com/prettier/prettier/issues/12209#issuecomment-1028775466).

See https://github.com/prettier/prettier/blob/b62b8e6d414d10b7cb37c343012a3a89a38aad8a/scripts/vendors/README.md for architecture.

In this PR, bundles following 3 packages:

- `string-width`
- ~~`globby`~~(`globby` is replaced with `fast-glob` in our source code)
- `mem`

## Checklist

- [x] I’ve added tests to confirm my change works.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Comment by fisker** [comment]
Can we do this in `postinstall` script?

File: scripts/vendors/vendor-versions.mjs [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import fs from "node:fs/promises";
+import path from "node:path";
+import createEsmUtils from "esm-utils";
+
+const { __dirname } = createEsmUtils(import.meta);
+
+const vendorVersionsPath = path.join(__dirname, "vendor-versions.json");
```
Comment (fisker) [inline scripts/vendors/vendor-versions.mjs:7]:
```suggestion
const vendorVersionsPath = new URL("vendor-versions.json", import.meta.url);
```

**Comment by duailibe** [comment]
> Can we do this in `postinstall` script?

My 2 cents, I personally don't like putting stuff in `postinstall`

File: scripts/vendors/vendor-versions.mjs (hunk 1 above)
Comment (sosukesuzuki) [reply scripts/vendors/vendor-versions.mjs]: thanks [0f57999](https://github.com/prettier/prettier/pull/12217/commits/0f57999361f771d00492b2ce960d54f254a44a68)

**Comment by sosukesuzuki** [comment]
We don't have many opportunities to run this script, so I don't think it's worth running it in `postinstall`. We can also set up a `postinstall` later if we feel it is necessary.

File: scripts/clean-changelog-unreleased.mjs [hunk 2]
```diff
@@ -2,13 +2,13 @@
...
 import { fileURLToPath } from "node:url";
-import globby from "globby";
+import { globbySync } from "globby";
 
 const changelogUnreleasedDir = fileURLToPath(
   new URL("../changelog_unreleased", import.meta.url)
 );
 
-const files = globby.sync(["blog-post-intro.md", "*/*.md"], {
+const files = globbySync(["blog-post-intro.md", "*/*.md"], {
```
Comment (fisker) [inline scripts/clean-changelog-unreleased.mjs:11]: Globby accepts URL as `cwd` now.

Comment (fisker) [reply scripts/clean-changelog-unreleased.mjs:11]: Actually this is the reason I propose https://github.com/sindresorhus/globby/issues/176

**Review by fisker — APPROVED** [review]

**Pull request merged**